zebra/label_manager.c

```diff
@@ -96,40 +142,14 @@ int zread_relay_label_manager_request(int cmd, struct zserv *zserv)
 	if (cmd == ZEBRA_RELEASE_LABEL_CHUNK)
 		return 0;
 
-	/* read response */
-	src = zclient->ibuf;
-	dst = zserv->obuf;
-
-	stream_reset(src);
-
-	u_int16_t size;
-	u_char marker;
-	u_char version;
-	vrf_id_t vrf_id;
-	u_int16_t resp_cmd;
-	ret = zclient_read_header(src, zclient->sock, &size, &marker, &version,
-				  &vrf_id, &resp_cmd);
-	if (ret < 0) {
-		zlog_err("%s: Error reading label chunk response: %s", __func__,
-			 strerror(errno));
+	/* wait for response */
+	FD_ZERO (&readfds);
+	FD_SET (zclient->sock, &readfds);
+	select (zclient->sock+1, &readfds, NULL, NULL, NULL);
```

Comment:
We need it because in this PR I made the socket non-blocking.
The thread model was broken when we decided (in one of our Tuesday meetings) to do this synchronously (it was done async at the very beginning).
Before it was blocking on zclient_read_header, so I don't see a big difference here.